js/lbank2.js

```diff
@@ -396,26 +396,27 @@ module.exports = class lbank2 extends Exchange {
             }
         } else {
             this.checkRequiredCredentials ();
-            const query = this.keysort (this.extend ({
+            const timestamp = Date.now ();
+            const echostr = this.uuid22 () + this.uuid16 ();
+            const authParameters = this.rawencode (this.keysort (this.extend ({
                 'api_key': this.apiKey,
-            }, params));
-            const queryString = this.rawencode (query);
-            const message = this.hash (this.encode (queryString)).toUpperCase ();
-            const cacheSecretAsPem = this.safeValue (this.options, 'cacheSecretAsPem', true);
-            let pem = undefined;
-            if (cacheSecretAsPem) {
-                pem = this.safeValue (this.options, 'pem');
-                if (pem === undefined) {
-                    pem = this.convertSecretToPem (this.secret);
-                    this.options['pem'] = pem;
-                }
-            } else {
-                pem = this.convertSecretToPem (this.secret);
-            }
-            const sign = this.binaryToBase64 (this.rsa (message, this.encode (pem), 'RS256'));
+                'echostr': echostr,
+                'signature_method': 'HmacSHA256',
+                'timestamp': timestamp,
+            }, params)));
+            const message = this.hash (authParameters).toUpperCase ();
+            const sign = this.hmac (message, this.secret);
```

Comment:
you need to do `this.hmac (this.encode (message), this.encode (this.secret))` for it to work in python

Reply:
okay thanks will add that now